addons/mail/static/tests/core/record_tests.js

```diff
@@ -260,6 +260,22 @@ QUnit.test("Trusted insert on html field with { html: true }", async (assert) =>
     assert.strictEqual(world.body, "<p>world</p>");
 });
 
+QUnit.test("(Un)trusted insertion is applied even with same field value", async (assert) => {
+    (class Message extends Record {
+        static id = "id";
+        id;
+        body = Record.attr("", { html: true });
+    }).register();
+    const store = await start();
+    const rawMessage = { id: 1, body: "<p>hello</p>" };
+    let message = store.Message.insert(rawMessage);
```

Comment:
This is not great though, I feel like untrusted write on trusted field should almost raise an exception, as there's 99.99% chance this is a programming error and not the indented behavior.